index.d.ts

```diff
@@ -106,628 +100,3416 @@ interface ClientOptions {
   }
 }
 
-declare class Client extends EventEmitter {
+declare class Client<RequestBody = any, Response = any, Context = unknown> extends EventEmitter {
   constructor(opts?: ClientOptions);
   connectionPool: ConnectionPool;
   transport: Transport;
   serializer: Serializer;
   extend: ClientExtends;
-  child(opts?: ClientOptions): Client;
+  child(opts?: ClientOptions): Client<RequestBody, Response, Context>;
   close(callback?: Function): Promise<void> | void;
   /* GENERATED */
   async_search: {
-    delete: ApiMethod<RequestParams.AsyncSearchDelete>
-    get: ApiMethod<RequestParams.AsyncSearchGet>
-    submit: ApiMethod<RequestParams.AsyncSearchSubmit>
+    delete<Response = BodyType, Context = unknown>(): Promise<ApiResponse<Response, Context>>
```

Comment:
Personally a fan of the following naming rule for generics:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/generic-type-naming.md

This makes them stick out more e.g `TResponse` `TContext` and `TRequest`.